app/models/host/managed.rb

```diff
@@ -434,15 +434,13 @@ def info
     info_hash
   end
 
-  # JSON is auto-parsed by the API, so these should be in the right format
-  def self.import_host_and_facts(hostname, facts, certname = nil, proxy_id = nil)
+  def self.import_host(hostname, import_type, certname = nil, proxy_id = nil)
```

Comment:
I wanted to avoid it, you should always specify what you're trying to import. Take it as a step towards the state where puppet is just another cfgmgmt tool we support and it can be extracted to a plugin.